databricks/koalas/series.py

```diff
@@ -747,6 +747,44 @@ def unique(self):
         sdf = self.to_dataframe()._sdf
         return _col(DataFrame(sdf.select(self._scol).distinct()))
 
+    def nunique(self, dropna: bool = True, approx: bool = False) -> int:
+        """
+        Return number of unique elements in the object.
+
+        Excludes NA values by default.
+
+        Parameters
+        ----------
+        dropna : bool, default True
+            Don’t include NaN in the count.
+        approx: bool, default False
+            If False, will use the exact algorithm and return the exact number of unique.
+            If True, it uses Spark's approximate algorithm, which is faster in most circumstances.
+            Note: this parameter is specific to Spark and is not found in pandas.
+
+        Returns
+        -------
+        The number of unique values as an int.
+
+        Examples
+        --------
+        >>> ks.Series([1, 2, 3, np.nan]).nunique()
+        3
+
+        >>> ks.Series([1, 2, 3, np.nan]).nunique(dropna=False)
+        4
+        """
+        sdf = self.to_dataframe()._sdf
```

Comment:
is there a way to reduce code duplication by simply calling the dataframe version of this series and just take the first number?

Reply:
basically i worry as the logic becomes more complicated (which it already is), it will become difficult to maintain.

Reply:
Indeed it does :) I've implemented your proposed change with d80f6a9.